### ISSUE TITLE
The child node "db_driver" at path "fos_user" must be configured.

### ISSUE THREAD

**Issue by NelsonSR** [kevinpapst/kimai2#947]
**Describe the bug**
I'm stuck in the middle of the installation process. After a while of execute
``php composer.phar install --no-dev --optimize-autoloader``
it stop with this error message:
```
Executing script cache:clear [KO]
 [KO]
Script cache:clear returned with error code 1
!!  
!!  In ArrayNode.php line 228:
!!                                                                       
!!    The child node "db_driver" at path "fos_user" must be configured.  
!!                                                                       
!!  
!!  
Script @auto-scripts was called via post-install-cmd
```
The weird thing is that `ArrayNode.php` doesn't exist and is not even called inside of any file in the `/var/www/kimai2` folder. 

The `db_driver` node mentioned only exists in the `/var/www/kimai2/config/packages/fos_user.yaml` config file and his value is `orm`.

**Desktop/Smartphone**
Please complete the following information if this seems to be a frontend issue:
 - Device: Lenovo N 500 laptop
 - OS: Debian GNU/Linux 10 «Buster»
 - Browser Firefox Quantum ESR
 - Version 60.7.2esr

**Additional context**
 - Kimai version 1.0.1
 - PHP 7.3.4-2 (cli)
 - Composer version 1.8.6 2019-06-11 15:03:05

**Comment by kevinpapst** [comment]
As this config is set:
https://github.com/kevinpapst/kimai2/blob/master/config/packages/fos_user.yaml#L2
it must be a local installation problem.
As you are using Debian Buster, so a brand day-one new system, I'd blame everything on its PHP system.

**Comment by NelsonSR** [comment]
> As this config is set:
> https://github.com/kevinpapst/kimai2/blob/master/config/packages/fos_user.yaml#L2
> it must be a local installation problem.
That `fos_user.yaml` file is exactly the same I have. Obviously, I haven't made any change in it, so both should be exactly the same.

> As you are using Debian Buster, so a brand new (day-one) system, I'd blame everything on its PHP installation ;-)
Yesterday I purged all the PHP modules and reinstall them. All the PHP 7.0 modules are gone, and just remains the PHP (no especific version) and PHP 7.3 modules. The `composer.phar` error message keep showing up.
 
> You can try to reinstall composer after `rm -rf vendor/` but I think the error will show up again afterwards.
I deleted `/usr/local/bin/composer.phar` and `/var/www/kimai2/composer.phar`, downloaded and reinstalled it, so I'm using the 1.8.6 version since two days ago.

**Comment by NelsonSR** [comment]
> No, I didn't mean reinstall composer, but run `composer install` AFTER deleting the vendor/ folder.
> And rm -rf var/cache/prod/* before as well.

`/var/www/kimai2/var/cache` is empty (there's only a .gitkeep inside)

**Comment by kevinpapst** [comment]
Nope, you just installed the dev packages. You did not delete the vendor/ before executing this.

**Comment by NelsonSR** [comment]
I don't know exactly what it was, but I did something that fix the issue now the error message is gone.
 
I thought it was because of coping `composer.phar` to `/usr/local/bin` but I deleted it and executed it again from the `kimai2` folder with no error message whatsover. 

I decide to keep going with the installation ignoring the error message. I step to the database installation just to stumble to another error message about `vendor/autoload.php`not found. Looking around I execute two commands (with no success):

`php composer.phar dump-autoload`
`php composer.phar update --no-scripts`

I return to previious error so I executed the script like I did yesterday, but coping the `composer.json` from the parent folder `/var/www` to `/var/www/kimai2` indicating to the script not to read from the parent folder. I noticed that instead of installing packages remove them. The next time I run composer all the package were installed:
```
$ composer install --no-dev --optimize-autoloader
Loading composer repositories with package information
Updating dependencies
Restricting packages listed in "symfony/symfony" to "~4.2.0"

Prefetching 4 packages 🎶 💨
  - Downloading (100%)

Package operations: 136 installs, 0 updates, 0 removals
  - Installing fzaninotto/faker (v1.8.0): Loading from cache
  - Installing pclzip/pclzip (2.8.2): Loading from cache
  - Installing phpoffice/common (0.2.9): Loading from cache
  - Installing zendframework/zend-escaper (2.6.0): Loading from cache
  - Installing phpoffice/phpword (0.15.0): Loading from cache
  - Installing nikic/php-parser (v4.2.2): Loading from cache
  - Installing symfony/finder (v4.2.10): Loading from cache
  - Installing symfony/polyfill-mbstring (v1.11.0): Loading from cache
  - Installing psr/container (1.0.0): Loading from cache
  - Installing psr/cache (1.0.1): Loading from cache
  - Installing symfony/contracts (v1.1.5): Loading from cache
  - Installing symfony/console (v4.2.10): Loading from cache
  - Installing psr/log (1.1.0): Loading from cache
  - Installing beberlei/doctrineextensions (v1.2.2): Loading from cache
  - Installing symfony/routing (v4.2.10): Loading from cache
  - Installing symfony/polyfill-ctype (v1.11.0): Loading from cache
  - Installing symfony/filesystem (v4.2.10): Loading from cache
  - Installing symfony/debug (v4.2.10): Loading from cache
  - Installing symfony/http-foundation (v4.2.10): Loading from cache
  - Installing symfony/event-dispatcher (v4.2.10): Loading from cache
  - Installing symfony/http-kernel (v4.2.10): Loading from cache
  - Installing symfony/dependency-injection (v4.2.10): Loading from cache
  - Installing symfony/config (v4.2.10): Loading from cache
  - Installing symfony/var-exporter (v4.2.10): Loading from cache
  - Installing psr/simple-cache (1.0.1): Loading from cache
  - Installing symfony/cache (v4.2.10): Loading from cache
  - Installing symfony/framework-bundle (v4.2.10): Loading from cache
  - Installing doctrine/lexer (1.0.2): Loading from cache
  - Installing doctrine/annotations (v1.6.1): Loading from cache
  - Installing doctrine/reflection (v1.0.0): Loading from cache
  - Installing doctrine/event-manager (v1.0.0): Loading from cache
  - Installing doctrine/collections (v1.6.2): Loading from cache
  - Installing doctrine/cache (v1.8.0): Loading from cache
  - Installing doctrine/persistence (1.1.1): Loading from cache
  - Installing symfony/doctrine-bridge (v4.2.10): Loading from cache
  - Installing doctrine/instantiator (1.2.0): Loading from cache
  - Installing doctrine/dbal (v2.9.2): Loading from cache
  - Installing doctrine/inflector (v1.3.0): Loading from cache
  - Installing doctrine/common (v2.10.0): Loading from cache
  - Installing doctrine/orm (v2.6.3): Loading from cache
  - Installing doctrine/doctrine-cache-bundle (1.3.5): Loading from cache
  - Installing symfony/security-core (v4.2.10): Loading from cache
  - Installing jdorn/sql-formatter (v1.2.17): Loading from cache
  - Installing doctrine/doctrine-bundle (1.11.2): Loading from cache
  - Installing doctrine/data-fixtures (v1.3.2): Loading from cache
  - Installing doctrine/doctrine-fixtures-bundle (3.2.2): Loading from cache
  - Installing erusev/parsedown (1.7.3): Loading from cache
  - Installing willdurand/jsonp-callback-validator (v1.1.0): Loading from cache
  - Installing willdurand/negotiation (v2.3.1): Loading from cache
  - Installing symfony/templating (v4.2.10): Loading from cache
  - Installing friendsofsymfony/rest-bundle (2.5.0): Loading from cache
  - Installing paragonie/random_compat (v2.0.18): Loading from cache
  - Installing twig/twig (v2.11.3): Loading from cache
  - Installing symfony/validator (v4.2.10): Loading from cache
  - Installing symfony/twig-bridge (v4.2.10): Loading from cache
  - Installing symfony/twig-bundle (v4.2.10): Loading from cache
  - Installing symfony/inflector (v4.2.10): Loading from cache
  - Installing symfony/property-access (v4.2.10): Loading from cache
  - Installing symfony/security-http (v4.2.10): Loading from cache
  - Installing symfony/security-guard (v4.2.10): Loading from cache
  - Installing symfony/security-csrf (v4.2.10): Loading from cache
  - Installing symfony/security-bundle (v4.2.10): Loading from cache
  - Installing symfony/options-resolver (v4.2.10): Loading from cache
  - Installing symfony/polyfill-intl-icu (v1.11.0): Loading from cache
  - Installing symfony/intl (v4.2.10): Loading from cache
  - Installing symfony/form (v4.2.10): Loading from cache
  - Installing friendsofsymfony/user-bundle (v2.1.2): Loading from cache
  - Installing behat/transliterator (v1.2.0): Loading from cache
  - Installing gedmo/doctrine-extensions (v2.4.37): Loading from cache
  - Installing hoa/exception (1.17.01.16): Loading from cache
  - Installing hoa/event (1.17.01.13): Loading from cache
  - Installing hoa/consistency (1.17.05.02): Loading from cache
  - Installing hoa/visitor (2.17.01.16): Loading from cache
  - Installing hoa/ustring (4.17.01.16): Loading from cache
  - Installing hoa/protocol (1.17.01.14): Loading from cache
  - Installing hoa/zformat (1.17.01.10): Loading from cache
  - Installing hoa/iterator (2.17.01.10): Loading from cache
  - Installing hoa/compiler (3.17.08.08): Loading from cache
  - Installing hoa/regex (1.17.01.13): Loading from cache
  - Installing hoa/math (1.17.05.16): Loading from cache
  - Installing hoa/stream (1.17.02.21): Loading from cache
  - Installing hoa/file (1.17.07.11): Loading from cache
  - Installing jms/metadata (2.0.0): Loading from cache
  - Installing jms/serializer (3.1.1): Loading from cache
  - Installing jms/serializer-bundle (3.4.1): Loading from cache
  - Installing kevinpapst/adminlte-bundle (2.8.3): Loading from cache
  - Installing myclabs/deep-copy (1.9.1): Loading from cache
  - Installing setasign/fpdi (1.6.2): Loading from cache
  - Installing mpdf/mpdf (v7.1.9): Loading from cache
  - Installing webmozart/assert (1.4.0): Loading from cache
  - Installing phpdocumentor/reflection-common (1.0.1): Loading from cache
  - Installing phpdocumentor/type-resolver (0.4.0): Loading from cache
  - Installing phpdocumentor/reflection-docblock (4.3.1): Loading from cache
  - Installing zircote/swagger-php (2.0.14): Loading from cache
  - Installing exsyst/swagger (v0.4.1): Loading from cache
  - Installing symfony/property-info (v4.2.10): Loading from cache
  - Installing nelmio/api-doc-bundle (v3.4.0): Loading from cache
  - Installing nelmio/cors-bundle (1.5.6): Loading from cache
  - Installing markbaker/matrix (1.1.4): Loading from cache
  - Installing markbaker/complex (1.4.7): Loading from cache
  - Installing phpoffice/phpspreadsheet (1.8.2): Loading from cache
  - Installing sensio/framework-extra-bundle (v5.3.1): Loading from cache
  - Installing symfony/polyfill-php72 (v1.11.0): Loading from cache
  - Installing symfony/var-dumper (v4.2.10): Loading from cache
  - Installing symfony/web-profiler-bundle (v4.2.10): Loading from cache
  - Installing symfony/stopwatch (v4.2.10): Loading from cache
  - Installing symfony/profiler-pack (v1.0.4): Loading from cache
  - Installing symfony/yaml (v4.2.10): Loading from cache
  - Installing monolog/monolog (1.24.0): Loading from cache
  - Installing easycorp/easy-log-handler (v1.0.7): Loading from cache
  - Installing symfony/monolog-bridge (v4.2.10): Loading from cache
  - Installing symfony/monolog-bundle (v3.4.0): Loading from cache
  - Installing symfony/debug-bundle (v4.2.10): Loading from cache
  - Installing symfony/debug-pack (v1.0.7): Loading from cache
  - Installing symfony/dotenv (v4.2.10): Loading from cache
  - Installing symfony/expression-language (v4.2.10): Loading from cache
  - Installing symfony/maker-bundle (v1.12.0): Loading from cache
  - Installing zendframework/zend-eventmanager (3.2.1): Loading from cache
  - Installing zendframework/zend-code (3.3.1): Loading from cache
  - Installing ocramius/proxy-manager (2.1.1): Loading from cache
  - Installing doctrine/migrations (v2.1.0): Loading from cache
  - Installing doctrine/doctrine-migrations-bundle (v2.0.0): Loading from cache
  - Installing symfony/orm-pack (v1.0.6): Loading from cache
  - Installing symfony/serializer (v4.2.10): Loading from cache
  - Installing symfony/polyfill-intl-idn (v1.11.0): Loading from cache
  - Installing symfony/translation (v4.2.10): Loading from cache
  - Installing pagerfanta/pagerfanta (v2.1.2): Loading from cache
  - Installing white-october/pagerfanta-bundle (v1.2.4): Loading from cache
  - Installing twig/extensions (v1.5.4): Loading from cache
  - Installing symfony/asset (v4.2.10): Loading from cache
  - Installing symfony/webpack-encore-bundle (v1.6.2): Loading from cache
  - Installing symfony/process (v4.2.10): Loading from cache
  - Installing symfony/web-server-bundle (v4.2.10): Loading from cache
  - Installing symfony/swiftmailer-bundle (v3.2.8): Loading from cache
  - Installing swiftmailer/swiftmailer (v6.2.1): Loading from cache
  - Installing egulias/email-validator (2.1.9): Loading from cache
Writing lock file
Generating optimized autoload files
ocramius/package-versions:  Generating version class...
ocramius/package-versions: ...done generating version class
Executing script cache:clear [OK]
Executing script assets:install [OK]
Executing script assets:install public [OK]
```

I'd like to repeat the installation process in another kimai2 folder just to be sure.

**Comment by NelsonSR** [comment]
Well, I repeated the installation process (using the version in a zip file instead the git repository). Some issue with the Internet connection cause some errors but nothing serious. 

I install composer using the bash script in https://getcomposer.org/doc/faqs/how-to-install-composer-programmatically.md. 

I jumped by misktake to the database installation step, causing another error about `autoload.php`, but after executing `php composer.phar install --no-dev --optimize-autoloader` no error message showed up.

The rest of the installation steps finished successfully.

In this case there wasn't a `/usr/local/bin/composer`file, only a local script in the kimai2 folder, so the only thing I can think as a cause of the initial error was an old version of `/usr/local/bin/composer`.
